src/codecoverage/bot/code_coverage_bot/uploader.py

```diff
@@ -96,13 +96,40 @@ def check_codecov_job():
         return False
 
 
-def gcp_upload(path, data):
+def gcp(repository, revision, data):
     '''
-    Upload a data payload on Google Cloud Storage
+    Upload a grcov raw report on Google Cloud Storage
+    * Compress with gzip
+    * Upload on bucket using revision in name
+    * Copy as latest file
     '''
-    assert isinstance(path, str)
-    assert isinstance(data, (bytes, str))
+    assert isinstance(data, bytes)
+    bucket = gcp_bucket()
 
+    # Compress report
+    archive = gzip.compress(data)
+
+    # Upload archive
+    path = '{}/{}.json'.format(repository, revision)
+    blob = bucket.blob(path)
+    blob.upload_from_string(archive)
+
+    # Update headers
+    blob.content_type = 'application/json'
+    blob.content_encoding = 'gzip'
+    blob.patch()
+
+    logger.info('Uploaded {} on {}'.format(path, bucket))
+
+    # Copy as latest file in same bucket
```

Comment:
Nit: is there a way to create kind of a symlink, instead of copying?

Reply:
I thought of that too, and did not see anything related [in the doc](https://cloud.google.com/storage/docs/json_api/v1/objects).